app/models.py

```diff
@@ -1642,6 +1644,86 @@ def serialize(self):
 
         return serialized
 
+    def serialize_with_cost_data(self):
+        serialized = self.serialize()
+        serialized["cost_details"] = {}
+        serialized["cost_in_pounds"] = "0.00"
+        serialized["is_cost_data_ready"] = True
+
+        if self.notification_type == "sms":
+            return self._add_cost_data_for_sms(serialized)
+        elif self.notification_type == "letter":
+            return self._add_cost_data_for_letter(serialized)
+
+        return serialized
+
+    def _add_cost_data_for_sms(self, serialized):
+        if not self._is_cost_data_ready_for_sms():
+            serialized["is_cost_data_ready"] = False
+            serialized["cost_details"] = {}
+            serialized["cost_in_pounds"] = None
+        else:
+            serialized["cost_details"]["billable_sms_fragments"] = self.billable_units
+            serialized["cost_details"]["international_rate_multiplier"] = self.rate_multiplier
+            sms_rate = self._get_sms_rate()
+            serialized["cost_details"]["rate"] = str(sms_rate)
+            serialized["cost_in_pounds"] = str(self.billable_units * self.rate_multiplier * sms_rate)
```

Comment:
I don't think you explained badly, I see your point. 

I still think we should send the value as string, since we are sending it via json and the value is more stable that way.

Here is a thread on this in Stackoverflow: https://stackoverflow.com/questions/35709595/why-would-you-use-a-string-in-json-to-represent-a-decimal-number

(I should probably include it in the commit message if we go with this 😳 )

Happy to have a call about this (tomorrow?)

Reply:
Hmm... but you have a point we send rather simple and small numbers, and we do no operations on them, so maybe float is ok, too - let's chat.

Reply:
Outcome: 

Both string and float are valid options, and both have been used by big players APIs - for example Twilio uses a float, while Paypal uses a string.

Leo has slight preference for float, since it means fewer conversions on both ours and our users' sides, so let's go with that.

Reply:
This is changed now, and in the docs, too.